### DISCUSSION TITLE
LCD 240x120 UC1608 with STM32F103RCT6

### DISCUSSION THREAD

**Discussion by EEJerry** [olikraus/u8g2#2240] [Q&A]
Hi all !
i user communication STM32F103RCT6 with LCD 240X120 Driver UC1608, my user library U8g2 not communication.
does anyone know at which step i am getting the error?
![image](https://github.com/olikraus/u8g2/assets/141943684/08d0da61-0379-41d1-9814-b48eb4e2e9d9)
![image](https://github.com/olikraus/u8g2/assets/141943684/c6c54ac3-64ce-4c25-b460-2fd655a7c69f)
![image](https://github.com/olikraus/u8g2/assets/141943684/ba92fad7-6301-442d-bbd0-28ed5425ef27)
![image](https://github.com/olikraus/u8g2/assets/141943684/d94fe1eb-dedb-4d4d-9c3a-151f42f7cbce)
![image](https://github.com/olikraus/u8g2/assets/141943684/48683b26-6a3f-4b6c-adfa-f27d2f28c1e0)
![image](https://github.com/olikraus/u8g2/assets/141943684/8d0aeb78-2e6a-4b43-b1c9-700fa6fe0b03)

**Comment by EEJerry** [comment]
@olikraus Did you come up with any ideas? Thank you very much !

**Reply by olikraus** [reply]
Difficult. Maybe test with stm32duino first.

**Reply by EEJerry** [reply]
I check with osilo the SPI is not working.
the setup of the u8g2 library has not been successful.

**Reply by olikraus** [reply]
> I check with osilo the SPI is not working.

ok, but then i think this problem is not related to u8g2. 

Again I suggest to use stm32duino, because the SPI subsystem from there is known to work.

**Reply by EEJerry** [reply]
@olikraus  do you have user Skyper, or Gmail, my send project code, you have check for my, configuar firmware .

**Reply by olikraus** [reply]
U8g2 is open source. I do this in my spare time, which is very limited. 
I don't have time to review / fix any project.

**Reply by EEJerry** [reply]
![image](https://github.com/olikraus/u8g2/assets/141943684/b7410668-e1d4-4ca8-a817-e7ba6c383000)
![image](https://github.com/olikraus/u8g2/assets/141943684/8bc769b4-53f9-4891-bfc1-029381b10423)
![image](https://github.com/olikraus/u8g2/assets/141943684/fb2384c1-e0b5-423a-9e4d-756ef7b509a0)
i am switching to STM32 arduino, STM32F103C8T6 KIT, 4 wire SPI LCD communication mode.
Using an osilo meter to measure pulses, all pins have pulses, but the screen does not display data,

**Comment by olikraus** [comment]
The default behavior for u8g2 are low active CS signals. You might need to change this in the code, if your display requires active high CS (as stated in the schematic). What is the name of your constructor?

**Reply by EEJerry** [reply]
i user UC1608 for LCD.

**Reply by EEJerry** [reply]
Where can I adjust the logic of the CS pin?

**Reply by olikraus** [reply]
Which u8g2 constructor function did you use?

**Reply by EEJerry** [reply]
U8G2_UC1608_ERC240120_1_4W_SW_SPI u8g2(U8G2_R0, /* clock=*/ PA5, /* data=*/ PA7, /* cs=*/ PA4, /* dc=*/ PB0, /* reset=*/ PA6);
this is the function i use, it's soft spi.

**Reply by EEJerry** [reply]
![image](https://github.com/olikraus/u8g2/assets/141943684/d80c2d3c-1feb-4ea5-bd51-fa6bf2311a5b)
I have reversed the CS pin of SPI but the display still has no data.

**Reply by olikraus** [reply]
Oh, it is already active high for this constructor:
https://github.com/olikraus/u8g2/blob/7c3517921accfaa23a36896288df264a0c24ed88/csrc/u8x8_d_uc1608.c#L234-L235

So ideally you should see the correct CS level with your scope.

I also see, that I marked this constructor with "experimental". It may not work reliable.
Maybe also test with `U8G2_UC1608_ERC24064_1_4W_SW_SPI` instead. Also remember to change the contrast with "setContrast" or even better, run the ContrastTest example to see whether there is any output.

**Reply by olikraus** [reply]
Ok, the user here confirms, that the 240x128 constructor works: https://github.com/olikraus/u8g2/issues/817

**Reply by olikraus** [reply]
And another confirmation is here: https://github.com/olikraus/u8g2/issues/101

**Reply by EEJerry** [reply]
i tried the library changes, i measure the signal the pins are all pulsed.
but the screen still has no data displayed.

**Reply by EEJerry** [reply]
![image](https://github.com/olikraus/u8g2/assets/141943684/392f1e8a-0889-43db-89a6-c8a0d77adc23)
Thank you so much @olikraus . 
finally showing the data, I switched to soft SPI, 4-wire mode.
